src/value.c

```diff
@@ -418,9 +417,6 @@ RSValue *RS_ArrVal(RSValue **vals, uint32_t len) {
   v->arrval.vals = vals;
   v->arrval.len = len;
   v->arrval.staticarray = 0;
-  for (uint32_t i = 0; i < len; i++) {
```

Comment:
why???
either:
1. Assert that the array is empty
2. Use NewArrayEx(... in the single place this function is actually used.. - this is preferrable

Reply:
@mnunberg this is a decision of ownership there was a memory leak before this fix, I think its reasonable that this function will take ownership